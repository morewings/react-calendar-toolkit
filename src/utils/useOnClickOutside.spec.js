import React, {useRef} from 'react';
import {mount} from 'enzyme';
import useOnClickOutside from './useOnClickOutside';

const callback = () => {};

const Component = () => {
  const ref = useRef();
  useOnClickOutside(ref, callback);
  return <div ref={ref} />;
};

describe('useOnClickOutside', () => {
  it('renders', () => {
    const component = mount(<Component />);
    expect(component).toMatchSnapshot();
  });

  it('adds and removes event listener to the document', () => {
    const addEventListener = jest.spyOn(document, 'addEventListener');
    const removeEventListener = jest.spyOn(document, 'addEventListener');
    const component = mount(<Component />);
    expect(addEventListener).toHaveBeenCalledTimes(2);
    expect(addEventListener.mock.calls[0][0]).toBe('mousedown');
    expect(addEventListener.mock.calls[1][0]).toBe('touchstart');
    component.unmount();
    expect(removeEventListener).toHaveBeenCalledTimes(2);
    expect(removeEventListener.mock.calls[0][0]).toBe('mousedown');
    expect(removeEventListener.mock.calls[1][0]).toBe('touchstart');
    addEventListener.mockRestore();
    removeEventListener.mockRestore();
  });
});
