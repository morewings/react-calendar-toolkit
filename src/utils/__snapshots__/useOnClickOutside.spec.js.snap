// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOnClickOutside renders 1`] = `
<Component>
  <div />
</Component>
`;
